client/src/templates/Challenges/components/preview-portal.tsx

```diff
@@ -72,16 +86,14 @@ class PreviewPortal extends Component<PreviewPortalProps> {
       this.props.togglePane('showPreviewPortal');
     });
 
-    this.props.storePortalDocument(this.externalWindow?.document);
-
-    this.mainWindow?.addEventListener('beforeunload', () => {
-      this.externalWindow?.close();
-    });
+    this.props.storePortalWindow(this.externalWindow);
   }
 
   componentWillUnmount() {
-    this.externalWindow?.close();
-    this.props.removePortalDocument();
+    if (this.mainWindow.location.href.split('/')[3] !== 'learn') {
```

Comment:
```suggestion
    if (this.mainWindow.location.href.split('/')[3] !== 'learn') {
```
This is a little too broad. It only closes the window if we navigate away from `/learn` entirely. For example, it works if we go to `/settings`, but not to `/learn/responsive-web-design`.

Reply:
ive made it more specific